src/features/transactions-graph/components/transactions-graph.tsx

```diff
@@ -15,63 +19,89 @@ export function TransactionsGraph({ transactionsGraphData }: Props) {
   const verticalsCount = verticals.length
   const gridTemplateColumns = `minmax(${horizontalTitleWidth}px, ${horizontalTitleWidth}px) repeat(${verticalsCount}, ${graphConfig.colWidth}px)`
 
+  const visualRef = useRef<HTMLDivElement>(null)
+
+  const downloadImage = useCallback(async () => {
+    const visual = visualRef.current
+    if (!visual) return
+    const canvas = await html2canvas(visual, {
+      scale: 2,
+      onclone: (_, element) => {
+        element.style.padding = '10px'
+      },
+    })
+    const dataUrl = canvas.toDataURL()
+    const link = document.createElement('a')
+    link.href = dataUrl
+    link.setAttribute('download', 'transactions-visual.png')
+    // TODO: This approach won't work in Tauri, so we'll need to handle with Tauri's APIs
+    link.click()
+  }, [])
+
   return (
-    // Don't change this id value, it's used by a bot Alessandro is building.
-    <div id="visual-transactions">
```

Comment:
have you confirmed that the id isn't needed anymore?

Reply:
yes, only the id on the button will be used.